packages/profiles/test/musig/btc.musig.registration.test.ts

```diff
@@ -0,0 +1,292 @@
+/* eslint-disable jest/expect-expect */
+import "jest-extended";
+import "reflect-metadata";
+
+import { bootContainer, makeProfile } from "../mocking";
+import { IProfile, IReadWriteWallet } from "../../source/contracts";
+import nock from "nock";
+
+import { mockMusigServer } from "./musig.test.helpers";
+import { mnemonics as testMnemonics } from "../fixtures/identity";
+import { Signatories } from "@payvo/sdk";
+import { oneSignatureNativeSegwitMusigRegistrationTx } from "../../../btc/test/fixtures/musig-native-segwit-txs";
+
+jest.setTimeout(60_000);
+
+describe("BTC", () => {
+	let profile1, profile2, profile3: IProfile;
+
+	const { mockServerResponse, resetServerResponseMocks } = mockMusigServer({
+		url: "https://btc-test-musig.payvo.com",
+	});
+
+	beforeAll(() => {
+		bootContainer();
+
+		// Default mocks
+		nock("https://btc-test.payvo.com:443")
+			// .get("/api/blockchain")
+			// .reply(200, require("../fixtures/client/blockchain.json"))
+			// .get("/api/node/configuration")
+			// .reply(200, require("../fixtures/client/configuration.json"))
+			// .get("/api/peers")
+			// .reply(200, require("../fixtures/client/peers.json"))
+			// .get("/api/node/configuration/crypto")
+			// .reply(200, require("../fixtures/client/cryptoConfiguration.json"))
+			// .get("/api/node/syncing")
+			// .reply(200, require("../fixtures/client/syncing.json"))
+			// // Wallet mocks
+			// .get("/api/wallets/DABCrsfEqhtdzmBrE2AU5NNmdUFCGXKEkr")
+			// .reply(200, require("../fixtures/wallets/DABCrsfEqhtdzmBrE2AU5NNmdUFCGXKEkr.json"))
+			// .get("/api/wallets/DCX2kvwgL2mrd9GjyYAbfXLGGXWwgN3Px7")
+			// .reply(200, require("../fixtures/wallets/DCX2kvwgL2mrd9GjyYAbfXLGGXWwgN3Px7.json"))
+			// .get("/api/wallets/DDHk393YcsxTPN1H5SWTcbjfnCRmF1iBR8")
+			// .reply(200, require("../fixtures/wallets/DDHk393YcsxTPN1H5SWTcbjfnCRmF1iBR8.json"))
+			.persist();
+
+		profile1 = makeProfile({ id: "profile-id-1" });
+		profile2 = makeProfile({ id: "profile-id-2" });
+		profile3 = makeProfile({ id: "profile-id-3" });
+	});
+
+	afterEach(() => {
+		resetServerResponseMocks();
+	});
+
+	describe("MuSig Registration", () => {
+		it("should perform a 2 out of 2 registration", async () => {
+			// TODO We shouldn't need to create a wallet first, but DW requires so, as the button to create musig wallet is inside an existing wallet
```

Comment:
Please see TODO my comments in this test.

Reply:
What would the alternative be? Just always showing a button to create BTC MuSig wallets even if you don't have any interest in BTC? Requiring a wallet to see it might seem odd from a purely technical PoV because BTC doesn't require it but from a client and UX PoV it makes a lot more sense to require a wallet.

Reply:
I can't think of an alternative really. The originator wallet and the musig wallet only have in common the passphrase, that's it. Not a single public key (out of the infinite ones) in common.
If we still want them to be related, then that relationship should include the derivation schema and path for the musig wallet. With those two, then the original wallet can derivate the public key to check the musig server for transactions, or to see if a transaction has its signature pending.